feat: Implemented component for Folksonomy Editor

heymitali opened the pull request:
### What
- Implemented web components for Folksonomy Edit and Folksonomy Auth
- Did all the API integration with folksonomy APIs.
- Implemented functionalities for all buttons including Edit, Delete, Submit
- Dynamically updating the component on various actions. Like on Submit - new row should be added without page refresh.
- Added Authentication component for testing because auth token is needed for folksonomy APIs. [Please feel free to correct if any other way is preferred]
- Added links to all respective keys.
- Implemented Responsive Design.
- Added support for both view and edit page types for Product page in Editor component.
- Handled translations for all text added.

### Screenshot
Folksonomy Auth Component
Usage: ` <folksonomy-auth />`

![Screenshot 2025-04-11 at 3 47 28 AM](https://github.com/user-attachments/assets/707cf6e4-9290-427d-b4d9-33454ddc539e)
 
Folksonomy Editor (In View Page)
Usage: ` <folksonomy-editor product-id="6111035000430" page-type="view" />`

![Screenshot 2025-04-11 at 3 47 47 AM](https://github.com/user-attachments/assets/049782c1-602e-47aa-8130-e56e22924173)
 
Folksonomy Editor (In Edit Page)
Usage: ` <folksonomy-editor product-id="6111035000430" page-type="edit" />`

![Screenshot 2025-04-11 at 3 48 17 AM](https://github.com/user-attachments/assets/e7d59d65-c6c0-4ad7-bda9-de7686140464)
 
### Fixes bug(s)
- #43 

### Part of 
- #29

alexgarel commented:
thanks @heymitali !

I don't think we need the auth component, it's the parent app that will handle that. You just have to consider you can use the auth cookie.

@teolemon can you give your though on the interface.

teolemon commented:
- [ ] can you provide screenshots in responsive mode @heymitali ?
- [ ] I'd be in favor of trying to get a link to the property documentation page on the wiki (regardless of whether it exists or not, we have a fixed pattern) https://wiki.openfoodfacts.org/Folksonomy/Property/xxxxxxxx: https://wiki.openfoodfacts.org/Folksonomy/Property/conservation:shelf_life
- [ ] it would be nice to be able to sort and filter the table
- [ ] it would be nice to alternate row colors for better legibility
- [ ] do you have autocompletion for the key ?
- [ ] do you have autocompletion for the value (based on the key ?)
- [ ] does the key and value text wrap properly if it's very long ?
- [ ] folksonomy engine >> Folksonomy Engine
- [ ] Does the system perform intuitively in case of error (no network, logout…) ?
-  [ ] Do we have a confirmation step on clicking "Delete" ?

@heymitali @alexgarel that's just a long list that goes beyond this PR. We could have a reflexion on the buttons further down the line, but it globally works.

heymitali commented:
These things are already done:
- [x] can you provide screenshots in responsive mode ? (Done)
  - I have added screenshot in the original description of the PR, please check.

- [x]  I'd be in favor of trying to get a link to the property documentation page on the wiki (regardless of whether it exists or not, we have a fixed pattern) https://wiki.openfoodfacts.org/Folksonomy/Property/xxxxxxxx: https://wiki.openfoodfacts.org/Folksonomy/Property/conservation:shelf_life (Done)

  - I am already doing it. but i was using versel url initially in this PR. After your comment, I have updated the link.

- [x] it would be nice to alternate row colors for better legibility (Done)

- [x] does the key and value text wrap properly if it's very long ? (Already Done)

- [x] folksonomy engine >> Folksonomy Engine (Done)

- [x]  Do we have a confirmation step on clicking "Delete" ?

  - Yes, the current PR, already provides a confirmation button and only when user clicks on confirm button, then only the delete API will be triggered. I have added screenshot for this in the description, please check.

These things I was planning to do in separate PR, This will help in keeping PR short for easy review. @teolemon please confirm if this is okay?

- [ ]  Does the system perform intuitively in case of error (no network, logout…) ?
- [ ] it would be nice to be able to sort and filter the table
- [ ] do you have autocompletion for the key ?
- [ ] do you have autocompletion for the value (based on the key ?)
- [ ]  We could have a reflexion on the buttons further down the line, but it globally works. (More Information Needed)

heymitali commented on web-components/src/components/folksnomy/delete-modal.ts:
```diff
@@ -0,0 +1,115 @@
+    }
+    button.cancel:hover {
+      background-color: #bbb;
+    }
```
done

heymitali commented on web-components/src/components/folksnomy/folksonomy-editor-row.ts:
```diff
@@ -0,0 +1,415 @@
+      border-radius: 20px;
+      font-size: 0.9rem;
+      height: 1.8rem;
+    }
```
done, I have used those common css. I will make sure that I will do it this way in future developments.

heymitali commented on web-components/src/components/folksnomy/delete-modal.ts:
```diff
@@ -0,0 +1,115 @@
+      cursor: pointer;
+      transition: background-color 0.3s;
+    }
+    button:hover {
```
done

heymitali commented on web-components/src/components/folksnomy/folksonomy-editor.ts:
```diff
@@ -0,0 +1,259 @@
+                    empty
+                  />`}
+                </table>
+              </form>
```
done

heymitali commented on web-components/src/api/folksonomy.ts:
```diff
@@ -0,0 +1,129 @@
+    throw error
+  }
+}
+
```
done. I have added type file.

heymitali commented on web-components/src/components/folksnomy/folksonomy-editor.ts:
```diff
@@ -0,0 +1,259 @@
+   * @type {boolean}
+   */
+  @property({ type: String, attribute: "product-id" })
+  productId = ""
```
done

heymitali commented on web-components/src/api/folksonomy.ts:
```diff
@@ -0,0 +1,129 @@
+import { DEFAULT_FOLKSONOMY_CONFIGURATION } from "../constants"
+
+const BASE_URL = DEFAULT_FOLKSONOMY_CONFIGURATION.baseUrl
```
got it, I have fixed this.

heymitali commented on web-components/src/components/folksnomy/folksonomy-auth.ts:
```diff
@@ -0,0 +1,149 @@
+import { LitElement, html, css } from "lit"
```
done

heymitali commented:
@Kout95 @alexgarel I have resolved all the review comments, please check.

teolemon commented:
@heymitali feature wise, we can of course leave the rest of the points for another PR 👍

heymitali commented:
@Kout95 @teolemon @alexgarel please help in reviewing this PR, and let me know if any other changes are required. I have already resolved all the review comments and also fixed merged conflicts. Please check.

heymitali commented:
@Kout95 all comments are resolved. please check.
